Implement ShowGroupsCommand and ListGroupCommand

g-lilian opened the pull request:
Allow users to list all existing groups, and display flashcards from a specified group.

g-lilian committed:
Implement ShowGroupsCommand

Allow users to use "showgroups" to list all existing groups.

g-lilian committed:
Update text-ui-test for ShowGroupsCommand

g-lilian committed:
Implement ListGroupCommand

Users can use "list-group [groupName]" to view a list of flashcards in the group.

g-lilian committed:
Update text-ui-test for ListGroupCommand and handle exceptions

Zhilin-Huang reviewed (commented):
`ShowGroupsCommand` looks good to me, but I wonder a bit about the naming of the `ListGroupCommand`. When I first saw this name I thought it was for listing all the groups haha I'm thinking maybe we can consider naming it to `ListGroupFlashcardsCommand` or `ListFlashcardsInGroupCommand` but these names are a bit lengthy. What do you think?

Also in the current implementation we are identifying groups with their names directly and this might be a bit inconvenient for the users because they'll need to input the group name exactly as it is to list flashcards in a group. Maybe we can use ID to identify groups like how we did for flashcards in FlashcardList? This might require quite some change but it's less costly to do so at this stage than later.

g-lilian committed:
Rename ListGroupsCommand to ListFlashcardsInGroupCommand

g-lilian committed:
Modify getFlashcardsInGroup to return using both name and group index

Users can now use either list-group [groupName] or list-group [groupIdx] to show the flashcards in the group.

g-lilian committed:
Update text-ui-test and fix checkstyle

g-lilian committed:
Fix text-ui-test error

g-lilian commented:
> `ShowGroupsCommand` looks good to me, but I wonder a bit about the naming of the `ListGroupCommand`. When I first saw this name I thought it was for listing all the groups haha I'm thinking maybe we can consider naming it to `ListGroupFlashcardsCommand` or `ListFlashcardsInGroupCommand` but these names are a bit lengthy. What do you think?
> 
> Also in the current implementation we are identifying groups with their names directly and this might be a bit inconvenient for the users because they'll need to input the group name exactly as it is to list flashcards in a group. Maybe we can use ID to identify groups like how we did for flashcards in FlashcardList? This might require quite some change but it's less costly to do so at this stage than later.

Agreed, have made the changes. Now the user can use either name or index to display the flashcards in the group (will search for name first, if no name matches, then check for the index).

Zhilin-Huang reviewed (approved):
LGTM!
